### PR TITLE
add react native polyfill to getting started

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes:

Add instructions to include missing polyfill for react native getting started.

#### Related GitHub issue #, if available:

https://github.com/aws-amplify/amplify-js/issues/10764

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

            
### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by dpilch** [comment]
I edited the wrong files. This should be on the DataStore getting started doc. Please ignore until fixed.

**Comment by dpilch** [comment]
Fixed to use only the DataStore getting started pages now.

File: src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Add Polyfills
+
+React Native is missing some standard library funcationality that is required by DataStore.
```
Comment (timngyn) [inline src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx:3]: Small typo that's causing the spellcheck to fail:

```suggestion
React Native is missing some standard library functionality that is required by DataStore.
```

**Review by alharris-at — APPROVED** [review]

File: src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+### Add Polyfills
+
+React Native is missing some standard library functionality that is required by DataStore.
+Add the following to the entry point of your app.
+
+```js
+import 'core-js/full/symbol/async-iterator';
+```
```
Comment (david-mcafee) [inline src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx:8]: cc @renebrandel - wanted to get your thoughts on phrasing here.

**Review by david-mcafee — CHANGES_REQUESTED** [review]
Minor question.

File: src/fragments/lib/datastore/react-native/getting-started/30_platformIntegration.mdx [hunk 3]
```diff
@@ -68,7 +68,7 @@ To enable SQLite with DataStore for enhanced local database performance, follow
 expo init AmplifyDataStoreExpo
 cd AmplifyDataStoreExpo
 npx amplify-app@latest
-expo install aws-amplify @aws-amplify/datastore-storage-adapter expo-sqlite expo-file-system @react-native-community/netinfo @react-native-async-storage/async-storage 
+expo install aws-amplify @aws-amplify/datastore-storage-adapter expo-sqlite expo-file-system @react-native-community/netinfo @react-native-async-storage/async-storage
```
[the remark this answers is not in the document]
Comment (dpilch) [reply src/fragments/lib/datastore/react-native/getting-started/30_platformIntegration.mdx:71]: From my testing when using expo `Symbol.asyncIterator` is available so `core-js` is not needed.

**Review by david-mcafee — DISMISSED** [review]
Would like to get @renebrandel's opinion on `src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx`, but otherwise looks good to me!

**Review by svidgen — APPROVED** [review]

File: src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+### Add Polyfills
```
Comment (renebrandel) [inline src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx:1]: Add required polyfills

File: src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+React Native is missing some standard library functionality that is required by DataStore.
```
Comment (renebrandel) [inline src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx:3]: React Native is missing some JavaScript standard library functionality that DataStore depends on. Add the following to the entry point of your app to polyfill these missing functionalities:

**Review by renebrandel — CHANGES_REQUESTED** [review]
suggested phrasing updates

**Review by renebrandel — APPROVED** [review]

**Pull request merged**